feat: Improve insights overall loading state

rafaeelaudibert opened the pull request:
Our loading state for insights isn't very modern and it occupies a lot of space. This was prompted by it not fitting the Web Vitals UI, let's tweak it all around the app and see if it looks nicer like this.

https://github.com/user-attachments/assets/5a33cbf9-be98-4f4a-a012-4cc450b87d4b


## Does this work well for both Cloud and self-hosted?

Oh yeah

## How did you test this code?

Visually locally + snapshots

---

Closes #28172

rafaeelaudibert committed:
feat: Improve insights overall loading state

This isn't very modern and it occupies a lot of space. This was prompted by it not fitting the Web Vitals UI, let's tweak it all around the app and see if it looks nicer like this.

rafaeelaudibert commented:
Marked as WIP while I fix CI - it'll certainly break

rafaeelaudibert commented on frontend/__snapshots__/scenes-app-insights-error-empty-states--estimated-query-execution-time-too-long--dark.png:
Yuck, what did I do

rafaeelaudibert commented on frontend/__snapshots__/scenes-app-insights-error-empty-states--long-loading--dark.png:
This will be very flaky because of the random number generation, will isolate this

rafaeelaudibert committed:
fix: Revert some UI changes, these looked ugly

rafaeelaudibert commented on frontend/src/scenes/insights/EmptyStates/EmptyStates.scss:
```diff
@@ -1,50 +1,4 @@
-            margin: 0.5rem 0;
-        }
-    }
+/* TODO: Migrate this to a CSS variable (with a `bg-` Tailwind class) once @adamleithp finishes his Tailwind work */
+.insights-empty-state {
+    background-color: rgb(129 129 129 / 20%);
 }
```
@adamleithp I'm expecting us to be able to use some kind of ´bg-gray-600/50` class once your TW4 changes are live, are we getting that?

adamleithp replied:
The idea is that we don't let engineers write this kind of tailwind class; adding opacity on the fly promotes deviation from controlled colors and leads to potential contrast issues!

rafaeelaudibert commented on frontend/src/scenes/insights/EmptyStates/EmptyStates.stories.tsx:
```diff
@@ -148,3 +149,13 @@ LongLoading.parameters = {
 
 export const FunnelSingleStep: Story = createInsightStory(funnelOneStep as any)
+
+export const LoadingMessages: StoryFn = () => {
```
I'm not really sure why this didn't generate a snapshots, working on it

rafaeelaudibert committed:
tests: Disable waiting for selector

Or else the snapshot is never generated because the selector is never found

robbie-c reviewed (commented):
I'll let the more design-oriented people give their feedback about that part. It looks like greptile has a few good comments here

robbie-c commented on frontend/src/scenes/insights/EmptyStates/EmptyStates.tsx:
```diff
@@ -126,39 +201,46 @@ export function StatelessInsightLoadingState({
+                        <span>{humanFriendlyNumber(rowsRead || 0, 0)} </span>
+                        <span>
+                            {estimatedRows && estimatedRows >= rowsRead ? (
+                                <span>/ ${humanFriendlyNumber(estimatedRows)} </span>
```
Looks legit

rafaeelaudibert committed:
fix: Finally fix `LoadingMessages` snapshot not working

rafaeelaudibert commented on frontend/__snapshots__/scenes-app-insights-loadingmessages--loading-messages--dark.png:
@corywatilo This is the list of messages I (Claude) came up with, but let me know if you dislike any of them - or if you want me to add some more

rafaeelaudibert committed:
style: Remove gray background from unnecessary components

We only need the gray background while loading, let's remove it from the other ones

rafaeelaudibert committed:
fix: Remove scrolling bar

rafaeelaudibert committed:
style: Remove redundant copy

rafaeelaudibert commented on frontend/src/scenes/insights/EmptyStates/EmptyStates.scss:
```diff
@@ -1,50 +1,4 @@
-    }
+/* TODO: Migrate this to a CSS variable (with a `bg-` Tailwind class) once @adamleithp finishes his Tailwind work */
+.insights-empty-state {
+    background-color: rgb(129 129 129 / 20%);
```
I noted above why this is the case

rafaeelaudibert commented on frontend/src/scenes/insights/EmptyStates/EmptyStates.stories.tsx:
```diff
@@ -148,3 +149,13 @@ LongLoading.parameters = {
+        <ul>
+            {LOADING_MESSAGES.map((message) => (
+                <li key={message}>{message}</li>
+            ))}
```
This is just a test

rafaeelaudibert committed:
ci: Attempt to fix prettier

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

adamleithp commented on frontend/__snapshots__/scenes-app-insights-error-empty-states--long-loading--dark.png:
ah here it is, looks great

adamleithp reviewed (changes requested):
I think it could probably sit on the default background like before (less box inside a box, and also that progress bar remainder is kinda lost...), that's my only suggestion!

rafaeelaudibert commented on frontend/__snapshots__/scenes-app-insights-error-empty-states--long-loading--light.png:
The whole idea that @corywatilo suggested in #28172 is the box so that it looks more like a Skeleton with that gray background. The problem here might be that we're trying to add it to places it shouldn't be, thoughts? If the UI isn't ready for a Skeleton-ready container (too box-y, which I agree), then maybe we should just scratch this for now. Or maybe we should make that bg optional? Not sure

adamleithp replied:
Ah yea Cory does specifically [in a] "frame", ok fair enough, let's see how it goes! LGTM, it's definitely better than it was.

github-actions[bot] committed:
Update UI snapshots for `webkit` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)
